x-pack/plugins/security_solution/public/management/components/console/components/command_list.tsx

```diff
@@ -41,24 +77,205 @@ export const CommandList = memo<CommandListProps>(({ commands }) => {
+        if (current[0].helpGroupPosition !== undefined) {
+          // If it already exists just move it to the end
+          if (acc[current[0].helpGroupPosition]) {
+            acc[acc.length] = acc[current[0].helpGroupPosition];
+          }
+
+          acc[current[0].helpGroupPosition] = sortBy(current, 'helpCommandPosition');
+        } else if (current.length) {
+          acc.push(current);
+        }
+        return acc;
+      },
+      []
+    );
+  }, [commands]);
+
+  const getTableItems = useCallback(
+    (
+      commandsByGroup: CommandDefinition[]
+    ): Array<{
+      [key: string]: { name: string; about: React.ElementType | string };
+    }> => {
+      if (commandsByGroup[0].helpGroupLabel === HELP_GROUPS.supporting.label) {
+        return [...COMMON_ARGS, ...commandsByGroup].map((command) => ({
+          [commandsByGroup[0]?.helpGroupLabel ?? otherCommandsGroupLabel]: command,
+        }));
+      }
+      return commandsByGroup.map((command) => ({
+        [commandsByGroup[0]?.helpGroupLabel ?? otherCommandsGroupLabel]: command,
+      }));
+    },
+    [otherCommandsGroupLabel]
+  );
+
+  const getTableColumns = useCallback(
+    (commandsByGroup) => {
+      return [
+        {
+          field: commandsByGroup[0]?.helpGroupLabel ?? otherCommandsGroupLabel,
+          name: commandsByGroup[0]?.helpGroupLabel ?? otherCommandsGroupLabel,
+          render: (command: CommandDefinition) => {
+            const commandNameWithArgs = getCommandNameWithArgs(command);
+            return (
+              <StyledEuiFlexGroup alignItems="center">
+                <EuiFlexItem grow={1}>
+                  <EuiDescriptionList
+                    listItems={[
+                      {
+                        title: <EuiBadge>{commandNameWithArgs}</EuiBadge>,
+                        description: (
+                          <>
+                            <EuiSpacer size="s" />
+                            <EuiText color="subdued" size="xs">
+                              {command.about}
+                            </EuiText>
+                          </>
+                        ),
+                      },
+                    ]}
+                    data-test-subj={getTestId('commandList-command')}
+                  />
+                </EuiFlexItem>
+                {/* Show EuiButtonIcon if is a command */}
+                {command.RenderComponent && (
+                  <EuiFlexItem grow={false}>
+                    <EuiToolTip
+                      content={i18n.translate(
+                        'xpack.securitySolution.console.commandList.addButtonTooltip',
+                        { defaultMessage: 'Add to text bar' }
+                      )}
+                    >
+                      <EuiButtonIcon
+                        iconType="plusInCircle"
+                        aria-label={`updateTextInputCommand-${command.name}`}
+                        onClick={() => updateInputText(`${commandNameWithArgs} `)}
```

Comment:
Yeah, that's why I suggested modifying the `useCallback` slightly so that it returns an anonymous function without any parameters that dispatches the action instead of the current implementation.